index.js

```diff
@@ -1,30 +1,162 @@
-main(input, 30);
+
+const consume = async (msgHandler, errorHandler) => {
+    let message = await getMessage();
+    if (!message) {
+        console.log('No messages in queue');
+        consume(msgHandler, errorHandler);
```

Comment:
`consume` is called in every situation, this is fine, but it might be cleaner to put the whole thing in a loop
```
while(true) {
    let message = await getMessage();
    if (!message) {
        console.log('No messages in queue');
        continue;
    }

    try {
        await msgHandler(message);
    } catch (err) {
        errorHandler(err);
    }
}
```

in general avoid use a if and exit instead of if else whenever possible to avoid nested if, and cleaner logic. this is just a style suggestion

Reply:
That's valid, a while loop is definitely cleaner and I agree with the if + early pattern. I generally try to use it, must have overlooked that here.